### PR TITLE
Fix too-many-ancestors

### LINKED ISSUE #4166: false positive too-many-ancestors if class inherits MutableMapping
### Steps to reproduce

Given a file `a.py`:
```python
# pylint: disable=missing-docstring

from typing import Iterator, MutableMapping, Optional


class PylintTestMap(MutableMapping[str, str]):
    config: MutableMapping[str, str] = dict()

    def __getitem__(self, item: str) -> Optional[str]:
        return self.config.get(item)

    def __len__(self) -> int:
        return len(self.config)

    def __iter__(self) -> Iterator[str]:
        return iter(self.config)

    def __setitem__(self, key: str, value: str) -> None:
        self.config[key] = value

    def __delitem__(self, key: str) -> None:
        del self.config[key]
```

### Current behavior

Result of `pylint a.py`:
```
************* Module a
a.py:7:0: R0901: Too many ancestors (8/7) (too-many-ancestors)

------------------------------------------------------------------
Your code has been rated at 9.23/10 (previous run: 9.23/10, +0.00)
```

### Expected behavior
No warning about ancestors

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.2
astroid 2.5.1
Python 3.9.1
```

This behaviour has been introduced with pylint 2.7.2. pylint 2.7.1 does not show this warning.

### LINKED ISSUE #4415: too-many-ancestors when inheriting from collections.abc.MutableSequence
Simply inheriting from collections.abc.MutableSequence generates a too-many-ancestors refactoring warning.
Yes, it does have that many ancestors, but I would consider only classes from the stdlib that are actually inherited by user code to be counted towards max ancestors.
And yes, I can disable this error in this case, or increase the max ancestors count, but it feels wrong to do that when just using the collections hierarchy for its intended purpose.

### Steps to reproduce

For a minimal MutableSequence implemented, ancestor.py:

```python
from collections.abc import MutableSequence


class ItemSequence(MutableSequence):
    """Minimal MutableSequence."""
    def __getitem__(self, key):
        return key

    def __setitem__(self, key, value):
        _ = key, value

    def __delitem__(self, key):
        _ = key

    def insert(self, index, value):
        _ = index, value

    def __len__(self):
        return 1
```

pylint ancestor.py returns:
`ancestor.py:4:0: R0901: Too many ancestors (8/7) (too-many-ancestors)`

### Expected behavior

Inheriting from a stdlib class shouldn't generate a Refactor warning.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.7.4
astroid 2.5.2
Python 3.7.7 (default, Aug  8 2019, 00:00:00) 
[GCC 8.3.1 20190311 (Red Hat 8.3.1-3)]
```

### PR DESCRIPTION
**Pull request by cdce8p**
## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
Fix false-positive `too-many-ancestors` by ignoring pre-defined builtin classes when calculating the number of ancestors. The list isn't complete but should cover most cases.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Related Issue
Closes #4166
Closes #4415

### CODE REVIEW

File: tests/functional/t/too/too_many_ancestors.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
-too-many-ancestors:20:0:Iiii:Too many ancestors (9/7)
-too-many-ancestors:23:0:Jjjj:Too many ancestors (10/7)
+too-many-ancestors:21:0:Iiii:Too many ancestors (8/7)
```
Comment (Pierre-Sassoulas) [inline tests/functional/t/too/too_many_ancestors.txt:1]: This actually fixed that test case too, the correct number IS 8 !

**Review by Pierre-Sassoulas — APPROVED** [review]
Good job !

File: tests/functional/t/too/too_many_ancestors.txt (hunk 1 above)
Comment (cdce8p) [reply tests/functional/t/too/too_many_ancestors.txt:1]: Depends if you con't `object`, but it's certainly not false.

**Comment by SagaraBattousai** [comment]
I also get this error when inheriting from any normal class (which doesn't cause this error) and a Protocol (which while not necessary is nice to ensure we override the required functions). Whilst we only inherit directly from two classes we get too-many-ancestors 8/7.

This error can be reproduced by:
from setuptools.command.build import SubCommand
from setuptools.command.build_ext import build_ext

# Sub Command is a Protocol but we want to explicitly inherit
class Foo(build_ext, SubCommand):
  .....

**Comment by Pierre-Sassoulas** [comment]
@SagaraBattousai please open a new issue your comment on a closed issue is going to get lost.

**Pull request merged**